lib/engine/step/tracker.rb

```diff
@@ -204,16 +204,18 @@ def update_token!(action, entity, tile, old_tile)
         if old_tile.paths.empty? &&
             !tile.paths.empty? &&
             cities.size > 1 &&
-            (token = cities.flat_map(&:tokens).find(&:itself))
-          actor = entity.company? ? entity.owner : entity
-          @round.pending_tokens << {
-            entity: actor,
-            hexes: [action.hex],
-            token: token,
-          }
-          @log << "#{actor.name} must choose city for token"
-
-          token.remove!
+            !(tokens = cities.flat_map(&:tokens).select(&:itself)).empty?
```

Comment:
cities.any? { |c| c.tokens.any?  }

cities.flat_map(&:tokens).any?

Reply:
Neither of those will properly set the value of `tokens`.

Here's something lengthier than what Jen wrote but it passes over the list just once, and avoids using `any?`-- since we use avoid using `any?` to mean `.size > 0`, in some places, I think it would be weird to use it to check for non-nil here.

```
(tokens = cities.reduce([]) { |t, c| t.concat(c.tokens.compact) }).empty?
```

Reply:
Or for something a touch closer to the current code:

```
(tokens = cities.reduce([]) { |t, c| t.concat(c.tokens.select(&:itself)) }).empty?
```

Reply:
i’d don’t realize you were using a setter

couldn’t you do cities.flatmap(&tokenz).compact

Reply:
yeah that should work too

Reply:
```suggestion
            !(tokens = cities.flat_map(&:tokens).compact).empty?
```